Sources/Spezi/Dependencies/DependencyCollection.swift

```diff
@@ -19,6 +19,14 @@ public struct DependencyCollection: DependencyDeclaration {
     init(_ entries: AnyDependencyContext...) {
         self.init(entries)
     }
+    
+    /// Creates a ``DependencyCollection`` from a closure resulting in a single generic type conforming to the Spezi  ``Module``.
+    /// - Parameters:
+    ///   - type: The generic type resulting from the passed closure, has to conform to ``Module``.
+    ///   - singleEntry: Closure returning a dependency conforming to ``Module``, stored within the ``DependencyCollection``.
+    public init<Dependency: Module>(for type: Dependency.Type = Dependency.self, singleEntry: (() -> Dependency)? = nil) {
```

Comment:
The `defaultValue`/`singleEntry` parameter is only supplied a nil entry if it is declared as a optional `@Dependency` (see [Optional Dependencies](https://swiftpackageindex.com/stanfordspezi/spezi/documentation/spezi/module/dependency#Optional-Dependencies)). It is used here https://github.com/StanfordSpezi/Spezi/blob/092eabc50a3600d8a03b43ad0d2dcd02914b223f/Sources/Spezi/Dependencies/DependencyPropertyWrapper.swift#L83

Does it make sense to declare an optional dependency in a `DependencyCollection`? Currently, I would probably leaning on just not allowing this as it just complicates the design and is currently also not supported.

Reply:
That's definitely a fair point, thanks for noticing it!

As you already stated, an optional dependency within a `DependencyCollection` doesn't really make sense, therefore I would opt for not allowing this.